pkg/operation/botanist/namespaces.go

```diff
@@ -74,6 +76,17 @@ func (b *Botanist) DeploySeedNamespace(ctx context.Context) error {
 			metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.LabelBackupProvider, b.Seed.GetInfo().Spec.Backup.Provider)
 		}
 
+		// Label namespace to pin all control-plane pods of a shoot cluster to one zone
+		// if the seed has workers across different availability zones.
+		if _, ok := b.Seed.GetInfo().Labels[v1beta1constants.LabelSeedMultiZonal]; ok {
+			haAnnotationValues, haIsConfigured := b.Shoot.GetInfo().Annotations[v1beta1constants.ShootAlphaControlPlaneHighAvailability]
+
+			if (haAnnotationValues == v1beta1constants.ShootAlphaControlPlaneHighAvailabilitySingleZone && gardenletfeatures.FeatureGate.Enabled(features.HAControlPlanes)) ||
+				!haIsConfigured {
+				metav1.SetMetaDataLabel(&namespace.ObjectMeta, v1beta1constants.ShootZonePinning, "")
+			}
```

Comment:
That's probably out of scope of this PR (migration non-HA to HA shoots) since it requires more complexity since the PVCs of etcd are always bound to a single zone.

However, if the feature gate gets disabled, the label should probably be removed again from the namespace?